active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,591 @@
+    {
+        using(args.GetDeferral())
+        {
+            bool success = await UpdateUserSettings(User thisUsersID);
+            if(!success)
+            {
+                //We were not able to update the settings! Don't close the tip and display the reason why.
+                args.Cancel = true;
+                ShowLastErrorMessage();
+            }
+        }
+    }
+}
+```
+
+### Related articles 
+
+* [Dialogs and flyouts](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/index)
+
+## Recommendations
+* Tips are impermanent and should not contain information or options that are critical to the experience of an application. 
+* Try to avoid showing teaching tips too often. Teaching tips are most likely to each recieve individual attention when they are staggered throughout long sessions or across multiple sessions.    
+* Keep tips succint and their topic clear. Research shows users, on average, only read 3-5 words and only comprehend 2-3 words before deciding whether to interact with a tip. 
+
+## API Details 
+
+```c++ 
+enum TeachingTipCloseButtonKind
+{
+    Auto,
+    Header,
+    Footer,
+};
+
+enum TeachingTipCloseReason
+{
+    CloseButton,
+    LightDismiss,
+    Programmatic,
+};
+
+enum TeachingTipPlacementMode
+{
+    Auto,
+    Top,
+    Bottom,
+    Left,
+    Right,
+    TopEdgeAlignedRight,
+    TopEdgeAlignedLeft,
+    BottomEdgeAlignedRight,
+    BottomEdgeAlignedLeft,
+    LeftEdgeAlignedTop,
+    LeftEdgeAlignedBottom,
+    RightEdgeAlignedTop,
+    RightEdgeAlignedBottom,
+};
+
+enum TeachingTipHeroPlacementMode
+{
+    Top,
+    Bottom,
+    Auto,
+};
+
+runtimeclass TeachingTipClosedEventArgs
+{
+    TeachingTipCloseReason Reason{ get; };
+};
+
+runtimeclass TeachingTipClosingEventArgs
+{
+    TeachingTipCloseReason Reason{ get; };
+    Boolean Cancel;
+    Windows.Foundation.Deferral GetDeferral();
+};
+
+unsealed runtimeclass TeachingTipTemplateSettings : Windows.UI.Xaml.DependencyObject
+{
+    TeachingTipTemplateSettings();
+
+    Windows.UI.Xaml.Thickness TopRightHighlightMargin;
+    Windows.UI.Xaml.Thickness TopLeftHighlightMargin;
+
+    static Windows.UI.Xaml.DependencyProperty TopRightHighlightMarginProperty{ get; };
+    static Windows.UI.Xaml.DependencyProperty TopLeftHighlightMarginProperty{ get; };
+}
+
+unsealed runtimeclass TeachingTip : Windows.UI.Xaml.Controls.ContentControl
+{
+    TeachingTip();
+
+    String Title;
+    String Subtext;
+
+    Boolean IsOpen;
+
+    String ActionButtonText;
+    Windows.UI.Xaml.Style ActionButtonStyle;
+    Windows.UI.Xaml.Input.ICommand ActionButtonCommand;
+    Object ActionButtonCommandParameter;
+
+    TeachingTipCloseButtonKind CloseButtonKind;
+    String CloseButtonText;
+    Windows.UI.Xaml.Style CloseButtonStyle;
+    Windows.UI.Xaml.Input.ICommand CloseButtonCommand;
+    Object CloseButtonCommandParameter;
+
+    Windows.UI.Xaml.Thickness TargetOffset;
```

Comment:
Not sure if it was covered anywhere yet, but I think there should be a knob that controls the display of the "beak" independent of the placement target. I think having placement relative to an element is useful and should be encouraged over absolute positioning within the window because it avoids needing margins & such. Also in a XamlIslands world there's no longer a window that we can assume given the current thread so I think we should always require a target and if you want it to show relative to the window you can set the target to the root of the window/island you want to be relative to.

Reply:
@jevansaks This makes sense to me, especially with regard to XAML Islands. I will add this API and create an example.

Reply:
I have added this functionality to the spec with this [commit](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/072e77b15dd60b39a06faf7b0067f6acf7e3f449). Tagging @StephenLPeters for visibility.